libshvchainpack/src/chainpack/accessgrant.cpp

```diff
@@ -114,4 +116,81 @@ RpcValue UserLoginResult::toRpcValue() const
 	}
 	return RpcValue(std::move(m));
 }
+
+//================================================================
+// AccessGrant
+//================================================================
+AccessGrant::AccessGrant(int level, const std::string &access_)
+	: accessLevelInt(level)
+	, access(access_)
+{
+}
+
+std::optional<MetaMethod::AccessLevel> AccessGrant::accessLevel() const
+{
+	return MetaMethod::accessLevelFromInt(accessLevelInt);
+}
+
+AccessGrant::AccessGrant(AccessLevel level)
+	: accessLevelInt(static_cast<int>(level))
+{
+}
+
+AccessGrant AccessGrant::fromShv2Access(std::string_view shv2_access, int access_level)
+{
+	AccessGrant ret;
+	if (auto level = MetaMethod::accessLevelFromInt(access_level); level.has_value()) {
+		ret.accessLevelInt = static_cast<int>(level.value());
+	}
```

Comment:
Kinda weird IMHO, converting from int to `AccessLevel` and then back to int

Reply:
`MetaMethod::accessLevelFromInt()` is checked conversion form int to AccessLevel, so it is rather safe conversion from int and back. For example `64` will not be converted.

Reply:
I wasn't talking about safety, just the fact that you're not really using the conversion. But it really is all about not having AccessLevel as an int everywhere.